### ISSUE TITLE
FP16 Grouped convolution is too slow on Titan V

### ISSUE THREAD

**Issue by irishev** [pytorch/pytorch#10290]
I tested a ResNeXt29_8x64d model (CIFAR-10) and found iteration on a convolution modulelist is always much faster than grouped convolution. It seems grouped convolution doesn't utilize the tensor core.

### Code of grouped convolution

```
class Block(nn.Module):
    '''Grouped convolution block.'''
    expansion = 4

    def __init__(self, in_planes, cardinality=8, bottleneck_width=64, stride=1):
        super(Block, self).__init__()
        self.bwidth = bottleneck_width
        group_width = cardinality * bottleneck_width
        self.conv1 = nn.Conv2d(in_planes, group_width, kernel_size=1, bias=False)
        self.bn1 = nn.BatchNorm2d(group_width)
        self.conv2 = nn.Conv2d(group_width, group_width, kernel_size=3, stride=stride, padding=1, groups=cardinality, bias=False)
        self.bn2 = nn.BatchNorm2d(group_width)
        self.conv3 = nn.Conv2d(group_width, self.expansion*bottleneck_width, kernel_size=1, bias=False)
        self.bn3 = nn.BatchNorm2d(self.expansion*bottleneck_width)

        self.shortcut = nn.Sequential()
        if stride != 1 or in_planes != self.expansion*bottleneck_width:
            self.shortcut = nn.Sequential(
                nn.Conv2d(in_planes, self.expansion*bottleneck_width, kernel_size=1, stride=stride, bias=False),
                nn.BatchNorm2d(self.expansion*bottleneck_width)
            )

    def forward(self, x):
        out = F.relu(self.bn1(self.conv1(x)))
        out = F.relu(self.bn2(self.conv2(out)))
        out = self.bn3(self.conv3(out))
        out += self.shortcut(x)
        out = F.relu(out)
        return out
```

### Code of modulelist iteration
```
class Block(nn.Module):
    '''Grouped convolution block.'''
    expansion = 4

    def __init__(self, in_planes, cardinality=8, bottleneck_width=64, stride=1):
        super(Block, self).__init__()
        self.bwidth = bottleneck_width
        group_width = cardinality * bottleneck_width
        self.conv1 = nn.Conv2d(in_planes, group_width, kernel_size=1, bias=False)
        self.bn1 = nn.BatchNorm2d(group_width)
        self.conv2 = nn.ModuleList([nn.Conv2d(bottleneck_width, bottleneck_width, kernel_size=3, stride=stride, padding=1, bias=False) for _ in range(cardinality)])
        self.bn2 = nn.BatchNorm2d(group_width)
        self.conv3 = nn.Conv2d(group_width, self.expansion*bottleneck_width, kernel_size=1, bias=False)
        self.bn3 = nn.BatchNorm2d(self.expansion*bottleneck_width)

        self.shortcut = nn.Sequential()
        if stride != 1 or in_planes != self.expansion*bottleneck_width:
            self.shortcut = nn.Sequential(
                nn.Conv2d(in_planes, self.expansion*bottleneck_width, kernel_size=1, stride=stride, bias=False),
                nn.BatchNorm2d(self.expansion*bottleneck_width)
            )

    def forward(self, x):
        out = F.relu(self.bn1(self.conv1(x)))
        out = F.relu(self.bn2(torch.cat([m(out[:,n*self.bwidth:(n+1)*self.bwidth,:,:]) for n,m in enumerate(self.conv2)], 1)))
        out = self.bn3(self.conv3(out))
        out += self.shortcut(x)
        out = F.relu(out)
        return out
```

### Result

Grouped convolution

[loss : 1.78515625]: 391/391 [03:38<00:00]
[loss : 1.599609375]: 391/391 [03:29<00:00]
[loss : 1.0234375]: 391/391 [03:28<00:00]

Modulelist iteration

[loss : 1.8349609375]:  391/391 [01:53<00:00]
[loss : 1.6005859375]:  391/391 [01:48<00:00]
[loss : 1.5390625]: 391/391 [01:47<00:00]

### System Info
PyTorch version: 0.4.0a0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.4 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: GPU 0: TITAN V
Nvidia driver version: 390.77
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.1.2

Versions of relevant libraries:
[pip] msgpack-numpy (0.4.1)
[pip] numpy (1.14.2)
[pip] torch (0.4.0a0)
[pip] torchtext (0.2.1)
[pip] torchvision (0.2.0)
[conda] Could not collect

**Comment by ngimel** [comment]
Grouped convolutions on V100 have been implemented in cudnn, closing
